### PR TITLE
Fix parallel flow

### PR DESCRIPTION
**Pull request by Sokol3783**
1. Fix problem of running new runnables from main thread
2. refactor logic of shutdown
3. refactor test for more clearer behavior of class

### CODE REVIEW

File: src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java [hunk 1]
```diff
@@ -51,37 +45,27 @@ public ParallelFlowExecutorServiceImpl(ExecutorService threadPoolExecutor,
      */
     @Override
     public void execute() {
-        while (FLAG) {
-            threadPoolExecutor.execute(tasksFactory.createExecutionWorker(
-                    service, scenarioHandler.getScenario(), proxyHandler.getProxy()));
+
+        threadPoolExecutor.execute(this::runParallelExecute);
```
Comment (OlexTul) [inline src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java:49]: With this approach, you use one thread from **threadPoolExecutornot** not effectively (not to complete the task from Queue), but you use it for this:
image.

**Review by OlexTul — CHANGES_REQUESTED** [review]

File: src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java [hunk 2]
```diff
@@ -51,37 +45,27 @@ public ParallelFlowExecutorServiceImpl(ExecutorService threadPoolExecutor,
...
+    }
+
+    private void runParallelExecute() {
+        while (!threadPoolExecutor.isShutdown()){
+            threadPoolExecutor.execute(() -> service.execute(
+                    scenarioHandler.getScenario(),
+                    proxyHandler.getProxy()
+            ));
         }
     }
```
Comment (git-alexbondarenko) [inline src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java:60]: The code exhibits an unbounded task queue problem, leading to the accumulation of over 200 million queued tasks within a few seconds. New tasks are submitted to the ThreadPoolExecutor without rate limiting or control, resulting in a backlog of tasks that can potentially exhaust system resources and degrade performance. I suppose, there is no need to create another thread to runParallelExecute().
Screenshot_20231031_130613
Screenshot_20231031_133046
Screenshot_20231031_134209

File: src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java (hunk 1 above)
Comment (Sokol3783) [reply src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java:49]: Give me some more details why it isn't efficient? And can you also explain why do you have to use run of new thread in main thread?

File: src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java (hunk 2 above)
Comment (Sokol3783) [reply src/main/java/executor/service/service/impl/parallel/ParallelFlowExecutorServiceImpl.java:60]: Yes but it more complex problem, than it's looks like, and if this pull have accepted, I'm going to offer some ways of solutions to this problem.

**Pull request closed without merging**